code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -101,16 +101,7 @@
 	update_spooky_icon()
 
 /mob/living/simple_animal/revenant/proc/update_spooky_icon()
-	if(revealed)
-		if(draining)
-			icon_state = "revenant_draining"
-			return
-		if(notransform)
-			icon_state = "revenant_stun"
-			return
-		icon_state = "revenant_revealed"
-		return
-	icon_state = "revenant_idle"
+	icon_state = revealed ? "revenant_[notransform ? "[draining ? "draining" : "stun"]" : "revealed"]" : "revenant_idle"
```

Comment:
I shouldn't have done this anyway because it'd make it way harder to implement icon customization for admin shenanigans.